modules/telemetry/include/hephaestus/telemetry/telemetry.h

```diff
@@ -0,0 +1,64 @@
+//=================================================================================================
+// Copyright (C) 2023-2024 HEPHAESTUS Contributors
+//=================================================================================================
+
+#pragma once
+
+#include <type_traits>
+
+#include "hephaestus/serdes/serdes.h"
+#include "hephaestus/telemetry/sink.h"
+
+namespace heph::telemetry {
+
+class Telemetry {
+public:
+  static void registerSink(ITelemetrySink* sink);
+
+  // Define the function input variable
+  template <serdes::JSONSerializable DataT>
+  static void log(const std::string& component, const std::string& tag, const DataT& data,
+                  ClockT::time_point log_timestamp = ClockT::now());
+
```

Comment:
Given these two are called by the user, add documentation and example here. Looking at example.cpp, things are very clear, so you could just drag this info over.